src/Core/Form/IdentifiableObject/DataHandler/ProductImageFormDataHandler.php

```diff
@@ -72,14 +58,15 @@ public function __construct(
     public function create(array $data)
     {
         $uploadedFile = $data['file'] ?? null;
+
         if (!($uploadedFile instanceof UploadedFile)) {
             throw new FileUploadException('No file was uploaded', UPLOAD_ERR_NO_FILE);
         }
 
         $command = new AddProductImageCommand(
             (int) ($data['product_id'] ?? 0),
             $uploadedFile->getPathname(),
-            null !== $this->contextShopId ? ShopConstraint::shop($this->contextShopId) : ShopConstraint::shop($this->defaultShopId)
+            $data['shop_id'] ? ShopConstraint::shop((int) $data['shop_id']) : ShopConstraint::allShops()
```

Comment:
Ooooh, nice catch!!

Reply:
Suggestion in case the key is not present:
```suggestion
            !empty($data['shop_id']) ? ShopConstraint::shop((int) $data['shop_id']) : ShopConstraint::allShops()
```